python/pylib/pose.py

```diff
@@ -20,912 +16,1317 @@ def quatInit():
-
-    R[0,0] = -Smu * Cl
-    R[0,1] = -Sl
-    R[0,2] = -Cmu * Cl
-    R[1,0] = -Smu * Sl
-    R[1,1] = Cl
-    R[1,2] = -Cmu * Sl
-    R[2,0] = Cmu
-    R[2,1] = 0.0
-    R[2,2] = -Smu
-    return R
+    eul = np.empty(shape=(np.shape(latlon)[0], 3))
+    eul[:,0] =  0.0;
+    eul[:,1] = -latlon[:,0] - 0.5 * pi
+    eul[:,2] =  latlon[:,1]
+    qe2n = euler2quat(eul)
 
-def rotate_ecef2ned( latlon ):
-    R = np.zeros((3,3))
-    # Smu, Cmu, Sl, Cl
+    if array == 0:
+        qe2n = np.squeeze(qe2n)
+    return qe2n
 
-    Smu = sin(latlon[0])
-    Cmu = cos(latlon[0])
-    Sl = sin(latlon[1])
-    Cl = cos(latlon[1])
 
-    R[0,0] = -Smu * Cl
-    R[0,1] = -Smu * Sl
-    R[0,2] = Cmu
+# Compute rotation matrix from NED to ECEF at given latitude/longitude
+def rotmat_ned2ecef(latlon):
+    if len(np.shape(latlon)) == 1:
+        latlon = np.expand_dims(latlon, axis=0)
+        array = 0
+    else:
+        array = 1
+
+    Smu = sin(latlon[:,0])
+    Cmu = cos(latlon[:,0])
+    Sl  = sin(latlon[:,1])
+    Cl  = cos(latlon[:,1])
+
+    R = np.empty(shape=(np.shape(latlon)[0], 3, 3))
+    R[:,0,0] = -Smu * Cl
+    R[:,0,1] = -Sl
+    R[:,0,2] = -Cmu * Cl
+    R[:,1,0] = -Smu * Sl
+    R[:,1,1] =  Cl
+    R[:,1,2] = -Cmu * Sl
+    R[:,2,0] =  Cmu
+    R[:,2,1] =  0.0
+    R[:,2,2] = -Smu
+
+    if array == 0:
+        R = np.squeeze(R)
+    return R
 
-    R[1,0] = -Sl
-    R[1,1] = Cl
-    R[1,2] = 0.0
 
-    R[2,0] = -Cmu * Cl
-    R[2,1] = -Cmu * Sl
-    R[2,2] = -Smu
+# Compute rotation matrix from ECEF to NED at given latitude/longitude
+def rotmat_ecef2ned(latlon):
+    if len(np.shape(latlon)) == 1:
+        latlon = np.expand_dims(latlon, axis=0)
+        array = 0
+    else:
+        array = 1
+
+    Smu = sin(latlon[:,0])
+    Cmu = cos(latlon[:,0])
+    Sl  = sin(latlon[:,1])
+    Cl  = cos(latlon[:,1])
+
+    R = np.empty(shape=(np.shape(latlon)[0], 3, 3))
+    R[:,0,0] = -Smu * Cl
+    R[:,0,1] = -Smu * Sl
+    R[:,0,2] =  Cmu
+    R[:,1,0] = -Sl
+    R[:,1,1] =  Cl
+    R[:,1,2] =  0.0
+    R[:,2,0] = -Cmu * Cl
+    R[:,2,1] = -Cmu * Sl
+    R[:,2,2] = -Smu
+
+    if array == 0:
+        R = np.squeeze(R)
     return R
 
 
-# Convert body attitude in Euler angles relative to ENU to
-# body attitude relative to NED frame
+# Convert body attitude in Euler angles relative to ENU frame
+# to body attitude in Euler angles relative to NED frame
 def enu2nedEuler(eul_e2b):
-    q_n2e = euler2quat(np.array([np.pi, 0, np.pi/2]))
-    q_e2b = euler2quatArray(eul_e2b)
-    q_n2b = mul_Quat_Quat_Array(q_e2b, q_n2e)
-    eul_n2b = quat2eulerArray(quat)
+    q_n2e = euler2quat(np.array([pi, 0.0, 0.5*pi]))
+    q_e2b = euler2quat(eul_e2b)
+    q_n2b = mul_Quat_Quat(q_e2b, q_n2e)
+    eul_n2b = quat2euler(q_n2b)
     return eul_n2b
 
 
-#  * NED to Euler_t
+#  Compute body atitude as Euler angles (pitch and heading with zero roll)
+#  given body X unit vector NED cordinates
+#  TODO: this function needs a better name
 def nedEuler(ned):
```

Comment:
This function could benefit from a better name. I added description based on what I think it does (no prior description)